gusto/equations/prognostic_equations.py

```diff
@@ -142,9 +142,29 @@ def generate_mass_terms(self):
             :class:`LabelledForm`: a labelled form containing the mass terms.
         """
 
+        if self.active_tracers is None:
+            tracer_names = []
+        else:
+            tracer_names = [tracer.name for tracer in self.active_tracers]
+
         for i, (test, field_name) in enumerate(zip(self.tests, self.field_names)):
             prog = split(self.X)[i]
             mass = subject(prognostic(inner(prog, test)*dx, field_name), self.X)
+            for j, ind_tracer_name in enumerate(tracer_names):
```

Comment:
Very picky, but would this variable make more sense to be called just `tracer_name`?